File: prisma/refs/py/longest-repeating-substring-with-replacement.py
from collections import defaultdict


def character_replacement(s, k):
    count = defaultdict(int)
    l = max_freq = best = 0
    for r, c in enumerate(s):
        count[c] += 1
        max_freq = max(max_freq, count[c])
        while r - l + 1 - max_freq > k:
            count[s[l]] -= 1
            l += 1
        best = max(best, r - l + 1)
    return best

File: prisma/refs/longest-repeating-substring-with-replacement.ts
export function characterReplacement(s: string, k: number): number {
  const count = new Array<number>(26).fill(0);
  const A = "A".charCodeAt(0);
  let l = 0, maxFreq = 0, best = 0;
  for (let r = 0; r < s.length; r++) {
    maxFreq = Math.max(maxFreq, ++count[s.charCodeAt(r) - A]);
    while (r - l + 1 - maxFreq > k) {
      count[s.charCodeAt(l) - A]--;
      l++;
    }
    best = Math.max(best, r - l + 1);
  }
  return best;
}
